### ISSUE TITLE
Map of Strings?

### ISSUE THREAD

**Issue by tellisnz-shift** [jdiazcano/cfg4k#60]
Is it possible to take a property with a Map<String, String>/Properties? I've tried creating parsers etc, but I couldn't get it to use my parser, and it seems the BindingInvocationHandler just throws SettingNotFound when trying to use any of the maps methods.

I even tried just creating a property that was a ConfigObject type and turn it into a map myself, but it threw a `SettingNotFound: Setting 'asObject' not found`.

While I know most of my config and its types beforehand and can add them to the interface, I'd like to be able to supply dynamic/unknown pairs of config through. Is this possible?

**Comment by tellisnz-shift** [comment]
The closest I can get is to have a List<Pair<String, String>> and create a Pair parser that splits a list of strings on a delimiter, like this:
```
config:
  - key1|value1
  - key2|value2
```
for
```
interface AppConfig {
  val config: List<Pair<String, String>>
}
```
But this is obviously a pretty hacky workaround... I'd just want this to work:
```
config:
  key1: value1
  key2: value2
```
with
```
interface AppConfig {
  val config: Map<String, String>
}
```

**Comment by jdiazcano** [comment]
It should just work so it might be a bug, I'm trying to reproduce (with the new testing!)

As example, working with json: https://github.com/jdiazcano/cfg4k/blob/25a9cd518520c8bd1d8a244a0a12b0460553f69f/cfg4k-json/src/test/kotlin/com/jdiazcano/cfg4k/json/ConfigProviderTest.kt#L266

In case that I can't, could you tell which source, loader and provider that you are using?

**Comment by jdiazcano** [comment]
Yeah, I can set a Map like:     `val mymap: Map<String, String>`

And it can be bound and retrieved, could you create a small snippet with the bug?

**Comment by tellisnz-shift** [comment]
Argh, sorry, I realised I wasn't using latest - this is solved somewhere between 0.9.0 and 0.9.3. Thanks!
